app/src/main/java/com/example/forcoms/Change_username.java

```diff
@@ -2,63 +2,77 @@
-
-    // TODO: Rename parameter arguments, choose names that match
-    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-    private static final String ARG_PARAM1 = "param1";
-    private static final String ARG_PARAM2 = "param2";
+import com.example.forcoms.userentity.UserData;
+import com.example.forcoms.userentity.UserViewModel;
 
-    // TODO: Rename and change types of parameters
-    private String mParam1;
-    private String mParam2;
+public class Change_username extends Fragment {
 
     public Change_username() {
-        // Required empty public constructor
     }
 
-    /**
-     * Use this factory method to create a new instance of
-     * this fragment using the provided parameters.
-     *
-     * @param param1 Parameter 1.
-     * @param param2 Parameter 2.
-     * @return A new instance of fragment Change_username.
-     */
-    // TODO: Rename and change types and number of parameters
-    public static Change_username newInstance(String param1, String param2) {
-        Change_username fragment = new Change_username();
-        Bundle args = new Bundle();
-        args.putString(ARG_PARAM1, param1);
-        args.putString(ARG_PARAM2, param2);
-        fragment.setArguments(args);
-        return fragment;
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_change_username, container, false);
     }
 
+    View currentView;
+
     @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        if (getArguments() != null) {
-            mParam1 = getArguments().getString(ARG_PARAM1);
-            mParam2 = getArguments().getString(ARG_PARAM2);
-        }
-    }
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
 
+        EditText oldUsername=view.findViewById(R.id.username_old);
+        EditText newUsername=view.findViewById(R.id.username_new);
+        Button buttonChange=view.findViewById(R.id.username_button_change);
+
+        UserViewModel userViewModel = new ViewModelProvider(requireActivity()).get(UserViewModel.class);
+
+        currentView=view;
+        buttonChange.setOnClickListener(view1->{
+            String newUsernameValue = newUsername.getText().toString();
+            String confirmNewUsernameValue = oldUsername.getText().toString();
+
+            if (!TextUtils.isEmpty(newUsernameValue.trim())){
+                if (newUsernameValue.equals(confirmNewUsernameValue)){
+                    String oldUsernameValue = oldUsername.getText().toString();
+                    UserData loggedInUser = userViewModel.getLoggedInUser();
+
+                    if (oldUsernameValue.equals(loggedInUser.getUsername())) {
+                        loggedInUser.setPassword(newUsernameValue);
```

Comment:
Seharusnya menggunakan method `setUsername`
bukan method `setPassword`

Jadinnya:
```suggestion
                        loggedInUser.setUsername(newUsernameValue);
```

Terima kasih